### PR TITLE
[expo-screen-orientation] Update documentation

### PR DESCRIPTION
**Pull request by lukmccall**
# Why

Follow up to https://github.com/expo/expo/pull/6760

# TODO

- [x] update changelog

### CODE REVIEW

File: packages/expo-screen-orientation/ios/EXScreenOrientation/EXScreenOrientationModule.m [hunk 1]
```diff
@@ -66,9 +66,17 @@ - (void)setModuleRegistry:(UMModuleRegistry *)moduleRegistry
...
     UIInterfaceOrientationMask orientationMask = [EXScreenOrientationUtilities maskFromOrientation:orientation];
+    if (!orientationMask) {
+      return reject(@"ERR_SCREEN_ORIENTATION_INVALID_ORIENTATION_LOCK", @"Invalid screen orientation lock.", nil);
+    }
+                                                                         
     allowedOrientationsMask = allowedOrientationsMask | orientationMask;
   }
   
+  if (![EXScreenOrientationUtilities doesDeviceSupportOrientationMask:allowedOrientationsMask]) {
+    return reject(@"ERR_SCREEN_ORIENTATION_UNSUPPORTED_ORIENTATION_LOCK", @"This device does not support this orientation.", nil);
```
Comment (brentvatne) [inline packages/expo-screen-orientation/ios/EXScreenOrientation/EXScreenOrientationModule.m]:
```suggestion
    return reject(@"ERR_SCREEN_ORIENTATION_UNSUPPORTED_ORIENTATION_LOCK", @"This device does not support the requested orientation.", nil);
```

**Comment by brentvatne** [comment]
it's not touched in this pr but it would be good to fix while we're here - under Object Types and Function Types there are bullet points are indented within a heading, eg:

### `ScreenOrientation.PlatformOrientationInfo`

    - screenOrientationConstantAndroid (_integer_): A constant to set using the Android native [API](https://developer.android.com/reference/android/R.attr.html#screenOrientation). For example, in order to set the lock policy to [unspecified](https://developer.android.com/reference/android/content/pm/ActivityInfo.html#SCREEN_ORIENTATION_UNSPECIFIED), -1 should be passed in. (Android only)

on the github markdown parser this strangely results in the bullet points being marked as codeblocks, as you can see above. this makes it difficult to read the full description. we should remove the indentation.

**Review by brentvatne — CHANGES_REQUESTED** [review]

**Review by brentvatne — APPROVED** [review]

File: docs/pages/versions/unversioned/sdk/screen-orientation.md [hunk 2]
```diff
@@ -21,15 +21,32 @@ On both iOS and Android platforms, changes to the screen orientation will overri
 
 This API is pre-installed in [managed](../../introduction/managed-vs-bare/#managed-workflow) apps. It is not yet available for [bare](../../introduction/managed-vs-bare/#bare-workflow) React Native apps.
 
+### Warning
+
+This module needs additional configuration to work properly on iPads. Unfortunately, it will disable support for the split view.
```
Comment (sjchmiela) [inline docs/pages/versions/unversioned/sdk/screen-orientation.md]: Could you elaborate a bit on what does it mean to _work properly on iPads_, why doesn't this module _work properly_ by default, what are the advantages and disadvantages of adding `requireFullScreen: true` property?

File: docs/pages/versions/unversioned/sdk/screen-orientation.md [hunk 3]
```diff
@@ -21,15 +21,32 @@ On both iOS and Android platforms, changes to the screen orientation will overri
...
+
+```json
+"ios": {
+    "requireFullScreen": true,
+}
+```
+
+#### Bare workflow
+
+Tick the `Requires Full Screen` checkbox in XCode. It should be located under `Project Target > General > Deployment Info`.
```
Comment (sjchmiela) [inline docs/pages/versions/unversioned/sdk/screen-orientation.md]:
```suggestion
Tick the `Requires Full Screen` checkbox in Xcode. It should be located under `Project Target > General > Deployment Info`.
```

File: docs/pages/versions/unversioned/sdk/screen-orientation.md [hunk 4]
```diff
@@ -21,15 +21,32 @@ On both iOS and Android platforms, changes to the screen orientation will overri
 
 This API is pre-installed in [managed](../../introduction/managed-vs-bare/#managed-workflow) apps. It is not yet available for [bare](../../introduction/managed-vs-bare/#bare-workflow) React Native apps.
 
+### Warning
+
+This module needs additional configuration to work properly on iPads. Unfortunately, it will disable support for the split view.
+
+#### Managed workflow
+
+Open your app.json and add the following inside of the "expo" field:
```
Comment (sjchmiela) [inline docs/pages/versions/unversioned/sdk/screen-orientation.md]:
```suggestion
Open your `app.json` and add the following inside of the `"expo"` field:
```

**Review by sjchmiela — COMMENTED** [review]
👍

File: docs/pages/versions/unversioned/sdk/screen-orientation.md [hunk 5]
```diff
@@ -21,15 +21,32 @@ On both iOS and Android platforms, changes to the screen orientation will overri
 
 This API is pre-installed in [managed](../../introduction/managed-vs-bare/#managed-workflow) apps. It is not yet available for [bare](../../introduction/managed-vs-bare/#bare-workflow) React Native apps.
 
+### Warning
+
+The Apple added support for the `split view` mode to iPads in iOS 9. This changed how the screen orientation is handled by the system. Briefly, for the iOS, your iPad is always in the landscape mode unless you open two applicationes side by side. To use the `ScreenOrientation` module you need to disable support for this feature. For more information about the `split view` mode, check out [the official Apple documentation](https://support.apple.com/en-us/HT207582).
```
Comment (sjchmiela) [inline docs/pages/versions/unversioned/sdk/screen-orientation.md]:
```suggestion
Apple added support for _split view_ mode to iPads in iOS 9. This changed how the screen orientation is handled by the system. To put the matter shortly, for the iOS, your iPad is always in the landscape mode unless you open two applications side by side. In order to be able to lock screen orientation using this module you will need to disable support for this feature. For more information about the _split view_ mode, check out [the official Apple documentation](https://support.apple.com/en-us/HT207582).
```

File: docs/pages/versions/unversioned/sdk/screen-orientation.md [hunk 6]
```diff
@@ -21,15 +21,32 @@ On both iOS and Android platforms, changes to the screen orientation will overri
...
+
+#### Managed workflow
+
+Open your `app.json` and add the following inside of the `"expo"` field:
```
Comment (sjchmiela) [inline docs/pages/versions/unversioned/sdk/screen-orientation.md]: How about putting the `expo` to inside the json? so it looks kinda like:

```
{
  expo: {
    ...
    ios: {
      ...
      requireFullScreen: true
```

File: docs/pages/versions/unversioned/sdk/screen-orientation.md [hunk 7]
```diff
@@ -122,13 +120,18 @@ Returns a promise with `void` value, resolving when the orientation is set and r
 
 #### Error Codes
 
-- `ERR_SCREEN_ORIENTATION_INVALID_ORIENTATION_LOCK` - an invalid [`OrientationLock`](#screenorientationorientationlock) was passed in.
+- `ERR_SCREEN_ORIENTATION_INVALID_ORIENTATION_LOCK` - an invalid [`OrientationLock`](#screenorientationorientationlock) was passed in (**isOnly**).
```
Comment (sjchmiela) [inline docs/pages/versions/unversioned/sdk/screen-orientation.md]:
```suggestion
- `ERR_SCREEN_ORIENTATION_INVALID_ORIENTATION_LOCK` - an invalid [`OrientationLock`](#screenorientationorientationlock) was passed in (**iOS only**).
```

I guess

File: docs/pages/versions/unversioned/sdk/screen-orientation.md [hunk 8]
```diff
@@ -218,6 +233,8 @@ An enum whose values can be passed to the [`lockAsync`](#screenorientationlockas
 - **`OrientationLock.OTHER`** -- A platform specific orientation. This is not a valid policy that can be applied in [`lockAsync`](#screenorientationlockasyncorientationlock).
 - **`OrientationLock.UNKNOWN`** -- An unknown screen orientation lock. This is not a valid policy that can be applied in [`lockAsync`](#screenorientationlockasyncorientationlock).
 
+> **Note** `OrientationLock.ALL` and `OrientationLock.PORTRAIT` are invalid on devices which not support `OrientationLock.PORTRAIT_DOWN`.
```
Comment (sjchmiela) [inline docs/pages/versions/unversioned/sdk/screen-orientation.md]:
```suggestion
> **Note** `OrientationLock.ALL` and `OrientationLock.PORTRAIT` are invalid on devices which do not support `OrientationLock.PORTRAIT_DOWN`.
```

File: packages/expo-screen-orientation/README.md [hunk 9]
```diff
@@ -23,12 +23,31 @@ npm install expo-screen-orientation
...
+   Replace
+
+   ```objc
+   UIViewController *rootViewController = [UIViewController new];
+   ```
+
+   with:
+
+   ```objc
+   UIViewController *rootViewController = [[EXScreenOrientationViewController alloc] initWithDefaultScreenOrientationMask:UIInterfaceOrientationMaskPortrait]; // through parameter you can specify your default orientation mask.
```
Comment (sjchmiela) [inline packages/expo-screen-orientation/README.md]: Can we default to using `init]`(less decisions for the developer) and note that people can also set a default orientation mask with a parameter?

File: packages/expo-screen-orientation/README.md [hunk 10]
```diff
@@ -23,12 +23,31 @@ npm install expo-screen-orientation
...
+   ```
+
+   To get more information about available orientation masks, please check out [UIInterfaceOrientationMask](https://developer.apple.com/documentation/uikit/uiinterfaceorientationmask?language=objc)
+
+> **Note** if you are using the custom view controller, this controller needs to extend the `EXScreenOrientationViewController`.
```
Comment (sjchmiela) [inline packages/expo-screen-orientation/README.md]:
```suggestion
> **Note** if you are using a custom view controller, the controller will need to extend the `EXScreenOrientationViewController`.
```

File: packages/expo-screen-orientation/README.md [hunk 11]
```diff
@@ -23,12 +23,31 @@ npm install expo-screen-orientation
...
+   UIViewController *rootViewController = [UIViewController new];
+   ```
+
+   with:
+
+   ```objc
+   UIViewController *rootViewController = [[EXScreenOrientationViewController alloc] initWithDefaultScreenOrientationMask:UIInterfaceOrientationMaskPortrait]; // through parameter you can specify your default orientation mask.
+   ```
+
+   To get more information about available orientation masks, please check out [UIInterfaceOrientationMask](https://developer.apple.com/documentation/uikit/uiinterfaceorientationmask?language=objc)
```
Comment (sjchmiela) [inline packages/expo-screen-orientation/README.md]:
```suggestion
   For more information about available orientation masks check out [UIInterfaceOrientationMask](https://developer.apple.com/documentation/uikit/uiinterfaceorientationmask?language=objc)
```

File: packages/expo-screen-orientation/ios/EXScreenOrientation/EXScreenOrientationViewController.m [hunk 12]
```diff
@@ -13,6 +13,15 @@ @interface EXScreenOrientationViewController ()
 
 @implementation EXScreenOrientationViewController
 
+- (instancetype)init
+{
+  if (self = [super init]) {
```
Comment (sjchmiela) [inline packages/expo-screen-orientation/ios/EXScreenOrientation/EXScreenOrientationViewController.m]: Let's delegate to another initializer:

`return [self initWithDefaultScreenOrientationMask:UIInterfaceOrientationMaskPortrait]`

**Review by sjchmiela — APPROVED** [review]

**Pull request merged**